### DISCUSSION TITLE
Force remove a node when it's not physically accessible

### DISCUSSION THREAD

**Discussion by imhotep** [zwave-js/zwave-js-ui#4005] [General]
I tried pairing one of my devices and ended up pairing 3 unknown devices (likely my neighbor's). I am not able to remove them from my stick because I can't put them in exclusion mode (for obvious reasons). Is there a way to force remove them from the zwave stick. Either through zwave-js-ui, node-zwave-js or ANY other way? I even tried connecting the stick to z-wave PC Controller 5, marking the nodes as "Is failed" and then "remove failed" but no luck there either.

**Comment by robertsLando** [comment]
not sure but I think the new nvm edit tool may be perfect for this. @AlCalzone ?

**Comment by imhotep** [comment]
@robertsLando are you referring to [@zwave-js/nvmedit](https://www.npmjs.com/package/@zwave-js/nvmedit) ?

Would I create a JSON NVM back up, edit the JSON to remove the node and then restore? Or did you have something else in mind?

**Reply by kpine** [reply]
https://zwave-js.github.io/nvmtool/

Which is basically editing the JSON, but hopefully has some checks in place.

**Reply by AlCalzone** [reply]
Both work, the online tool is a bit less involved. Essentially you create an NVM backup, load it in the tool, remove the corresponding entries, save the modified NVM and restore it.

**Disclaimer:**  
If the nodes are actually part of the network, this will not remove them. Only the controller will forget them. They may still think they are part of the network, try to communicate and cause traffic.
This process should not be used if there is another way to remove nodes.

You'll want to remove:
- every object `{...}` in `sucUpdateEntries` where `nodeId` matches the nodes you want to remove
- further down in the `nodes` dict, remove the entries of the node IDs you want to remove, e.g. `"7": {...}` if you want to remove node 7
- in all other entries in the `nodes` dict, remove the node IDs you want to remove from the `neighbors`
- If any of the node IDs show up in the `repeaterNodeIDs` of the `lwr` or `nlwr` of any node, remove the entire entry (`lwr` or `nlwr`).

Make sure the result is valid JSON, so no trailing commas, etc. The tool checks whether all required entries are there, but it does not check whether it makes sense.

**Reply by imhotep** [reply]
Thanks for the detailed description!

> This process should not be used if there is another way to remove nodes.

@AlCalzone Is there another way to remove devices without putting them in exclusion mode? I don't know which neighbor has these devices and it's a bit strange to knock on random people's doors to ask them if I can put their z-wave devices in exclusion mode xD

The only way I can think of is to plug in the stick to a laptop (I only have mac OS though so hope zwavejs works?) far away from these devices and then for sure they won't respond to ping and be excluded.

**Reply by kpine** [reply]
Sometimes wrapping the controller in aluminum foil will prevent it from communicating. Doesn't always work.

Here's an idea I've never tried. If you have a 700 or 800 series controller you can change the region (frequency). E.g. if you are in US pick Europe or something else. Then it won't be possible for the controller to communicate with the devices.

In either case, ping the problem nodes and they should fail and can be removed with "Remove failed node". Then re-configure the region if you changed it.

**Reply by AlCalzone** [reply]
> Is there another way to remove devices without putting them in exclusion mode?

Only "remove failed node" which requires communication with the device to fail.

**Comment by imhotep** [comment]
@kpine I had tried the foil method and it didn't work. I even stuck the stick in a metal box and it still didn't work :/ Changing the region causes the serial API to restart in sequence so didn't work either.

**Reply by kpine** [reply]
I just tested the frequency method myself. It worked flawlessly.

I don't know what you mean by "causes the serial API to restart in sequence". Maybe your controller doesn't support changing the frequency?

**Reply by imhotep** [reply]
This is what I get when changing frequency. Obviously all the nodes are dead but then when I ping I get this

```
20:26:51.516 CNTRLR   Serial API restarted unexpectedly.
20:26:51.516 CNTRLR   Currently active command will be retried...
20:26:51.517 CNTRLR   Starting hardware watchdog...
20:26:53.927 CNTRLR   Serial API restarted unexpectedly.
20:26:53.928 CNTRLR   Currently active command will be retried...
20:26:53.929 CNTRLR   Starting hardware watchdog...
20:26:56.267 CNTRLR   Serial API restarted unexpectedly.
20:26:56.268 CNTRLR   Currently active command will be retried...
20:26:56.269 CNTRLR   Starting hardware watchdog...
20:26:58.777 CNTRLR   Serial API restarted unexpectedly.
20:26:58.777 CNTRLR   Currently active command will be retried...
20:26:58.778 CNTRLR   Starting hardware watchdog...
20:27:01.099 CNTRLR   Serial API restarted unexpectedly.
20:27:01.100 CNTRLR   Currently active command will be retried...
20:27:01.100 CNTRLR   Starting hardware watchdog...
20:27:03.638 CNTRLR   Serial API restarted unexpectedly.
20:27:03.638 CNTRLR   Currently active command will be retried...
20:27:03.639 CNTRLR   Starting hardware watchdog...
```

**Reply by AlCalzone** [reply]
You may want to disable the watchdog in the Z-Wave JS UI settings. On some controllers it is a bit buggy.

**Comment by imhotep** [comment]
Changing region, disabling the watchdog and waiting until all of the nodes fail to communicate seems to have done the trick! Thanks all!
